src/mainwindow.cpp

```diff
@@ -9321,10 +9320,19 @@ void MainWindow::on_actionSearch_text_on_the_web_triggered() {
         return;
     }
 
+    // handling the case in which the saved engine id
+    // has been removed
+
     QSettings settings;
-    int selectedSearchEngineIndex = settings.value("searchEngineUrl").toInt();
-    QString searchEngineUrl = Utils::Misc::getSearchEnginesVector()
-            [selectedSearchEngineIndex].second;
```

Comment:
this will crash the app if the index isn't found

Reply:
That's why it's removed

Reply:
ah, sorry. it's getting late....